backend/modules/parsers/audioparser.py

```diff
@@ -0,0 +1,125 @@
+import json
+from typing import Any, Dict, List
+
+import aiofiles
+import aiohttp
+from langchain.docstore.document import Document
+
+from backend.logger import logger
+from backend.modules.model_gateway.model_gateway import model_gateway
+from backend.modules.parsers.parser import BaseParser
+from backend.modules.parsers.unstructured_io import UnstructuredIoParser
+from backend.types import ModelConfig
+
+
+class AudioParser(BaseParser):
+    """
+    AsyncAudioParser is a parser class for extracting text from audio input.
+
+    {
+        ".mp3": {
+            "name": "AudioParser",
+            "parameters": {
+                "model_configuration": {
+                    "name" : "faster-whisper/Systran/faster-distil-whisper-large-v3"
+                },
+                "max_chunk_size": 2000
+            }
+        }
+    }
+
+    """
+
+    supported_file_extensions = [
+        ".flac",
+        ".mp3",
+        ".mp4",
+        ".mpeg",
+        ".mpga",
+        ".m4a",
+        ".ogg",
+        ".wav",
+        ".webm",
+    ]
+
+    def __init__(
+        self, *, model_configuration: ModelConfig, max_chunk_size: int = 2000, **kwargs
+    ):
+        """
+        Initializes the AsyncAudioParser object.
+        """
+        self.model_configuration = ModelConfig.model_validate(model_configuration)
+        self.audio_processing_svc = None
+        self.max_chunk_size = max_chunk_size
+        super().__init__(**kwargs)
+
+    async def initialize(self):
+        """
+        Initialize the audio processing service.
+        """
+        self.audio_processing_svc = (
+            await model_gateway.get_audio_model_from_model_config(
+                model_name=self.model_configuration.name
+            )
+        )
+
+    async def get_chunks(
+        self, filepath: str, metadata: Dict[Any, Any] | None, **kwargs
+    ) -> List[Document]:
+        """
+        Get the chunks of the audio file.
+        """
+        if not self.audio_processing_svc:
+            await self.initialize()
+
+        try:
+            parsed_audio_text = []
+
+            async with self.audio_processing_svc as svc:
+                response: aiohttp.ClientResponse = await svc.get_transcription(
+                    audio_file_path=filepath
+                )
+
+                try:
+                    async for line in response.content:
+                        line = line.strip()
+                        if line:
+                            data = json.loads(
+                                line.decode("utf-8").strip().split("data: ")[1]
+                            )["text"]
```

Comment:
see comment about moving this to audio service itself
https://github.com/truefoundry/cognita/pull/337/files#r1765351573

Reply:
this can just become

```
async for text_chunk in self.audio_processing_svc.get_transcription(...):
    parsed_audio_text.append(text_chunk)
```